src/com/google/javascript/jscomp/PeepholeReplaceKnownMethods.java

```diff
@@ -73,10 +74,15 @@ private Node tryFoldKnownMethods(Node subtree) {
       Node callTarget = checkNotNull(subtree.getFirstChild());
 
       if (NodeUtil.isGet(callTarget)) {
-        if (isASTNormalized() && callTarget.getFirstChild().isQualifiedName()) {
```

Comment:
I'm not sure I follow why this `isASTNormalized()` check is incompatible with the new optimization.

I would have thought that waiting for the AST to be normalized would work just as well with "Array" as it did with "Math". Doesn't it?

Reply:
I don't know either. I couldn't figure out why that method is used. Is this optimization called multiple times but only once when the AST is normalized?

The strange part is when I use that condition, there's never any change for any `Array.of` call and no errors. Since the optimization only cares about the function name, I thought it would be fine to remove to fix the problem.

Reply:
These passes are always run after normalization, but there is also a "late peephole passes" run that runs after denormalization when the names may no longer be unique.

I think that by that point, any calls to Array.of should have already been replaced, so it should be reasonable to backoff here (just in the very uncommon case that denormalization named another namespace to "Array").

Reply:
I see. Is there a problem with my tests then? I only changed it because [this commit](https://github.com/google/closure-compiler/pull/3573/commits/e6d3e055e6ee93e7a9d4b61a47900fcc1088b142) failed.

Reply:
Oh interesting.  Looking at the similar `Math.abs` case, for example, it looks like the unit tests need an explicit call to `enableNormalize();`

Does that work?

Reply:
That did it! Will it normalize automatically for [this test](https://github.com/google/closure-compiler/pull/3573/files#diff-a41f00968b9e5a04015c0b2ecaacfcffR8106), since it's an integration test?

Reply:
That's right.  In the integration tests, there's no way to avoid normalization.

Reply:
Great, so I think this is done if you approve.

Reply:
Looks good. Sorry it was slow to figure out the new process, but thanks for going through the whole code review process!

Importing with copybara now.

Reply:
Don't worry. This change wasn't urgent.

And thanks for being so helpful through it!